column_reader_go18.go

```diff
@@ -35,12 +36,16 @@ func (r *columnReader[T]) ReadRequired(values []T) (n int, err error) {
 	if r.offset < len(r.buffer) {
 		n = copy(values, r.buffer[r.offset:])
 		r.offset += n
+		r.remain -= n
 		values = values[n:]
 	}
-	if r.decoder == nil {
+	if r.remain == 0 || r.decoder == nil {
 		return n, io.EOF
 	}
 	d, err := r.class.decode(r.decoder, values)
+	if r.remain -= d; r.remain == 0 && err == nil {
```

Comment:
this is a dumb style thing but this is a lot to read for one line, i'd just do two lines 

```
d, err := ...
r.remain -= d
if r.remain == 0 && err == nil {
    ... 
}
```